selfdrive/ui/sidebar.cc

```diff
@@ -56,8 +55,8 @@ static void ui_draw_sidebar_network_type(UIState *s) {
   const char *network_types[6] = {"--", "WiFi", "2G", "3G", "4G", "5G"};
   char network_type_str[32];
 
-  if (s->scene.networkType <= 5) {
-    snprintf(network_type_str, sizeof(network_type_str), "%s", network_types[s->scene.networkType]);
+  if ((int)s->scene.networkType <= 5) {
```

Comment:
Similar comment to below on the casts if there's a nice way to remove this one. We shouldn't make assumptions like "5"

Reply:
Seriously， we should not use this enum value as index of array.

from the capnproto's document:

> Keep in mind when writing switch blocks that an enum read off the wire may have a numeric value that is not listed in its definition. This may be the case if the sender is using a newer version of the protocol, or if the message is corrupt or malicious.

Reply:
can we use a std::map?

Reply:
sure we can. but I think switch is clear enough and is the "standard" way to do this type of thing.
use std::map can redule a few lines, but it  alloc memory from heap,and do a "hash table  search"  everytime to get the value from key.I really don't like that.

Reply:
Actually from a perf perspective I'm not sure std::map would be worse. switch statement = branch mispredict, idk if map would.

Reply:
@quillford you'r right.std::map is much clearer than switch.